lib/CL/pocl_llvm_api.cc

```diff
@@ -137,7 +142,112 @@ static llvm::sys::Mutex kernelCompilerLock;
 /* Global pocl device to be used by passes if needed */
 cl_device_id currentPoclDevice = NULL;
 
-static void InitializeLLVM();
+static std::map<cl_device_id, llvm::TargetMachine *> target_machines;
+static std::map<cl_device_id, PassManager *> kernel_passes_noflatten;
```

Comment:
This is because the inline method (all vs globals only) is decided per-module not per-device, depending on whether the module uses images. Once we get rid of our inline-all hack and let LLVM do the inlining, we can use just one pass manager list per device.